backend/elasticsearch-client/jdk-rest-client/src/main/java/org/hibernate/search/backend/elasticsearch/client/jdk/impl/ClientJdkHttpRequestInterceptor.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.backend.elasticsearch.client.jdk.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLDecoder;
+import java.net.http.HttpRequest;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.hibernate.search.backend.elasticsearch.client.common.spi.ElasticsearchRequestInterceptor;
+import org.hibernate.search.backend.elasticsearch.client.common.spi.ElasticsearchRequestInterceptorContext;
+
+
+record ClientJdkHttpRequestInterceptor(ElasticsearchRequestInterceptor elasticsearchRequestInterceptor)
+		implements HttpRequestInterceptor {
+
+	// https://docs.oracle.com/en/java/javase/25/docs/api/java.net.http/module-summary.html
+	//
+	// Host header is one of the restricted ^ so we skip it here:
+	private static final Set<String> HEADERS_TO_IGNORE = Set.of( "host" );
```

Comment:
some of those JDK client properties can only be set as system properties ... and what "restricted" headers to allow is one of those configs. 
I've tested the aws signing with the host header skipped and it seems to work ok, so I'll leave it as this for now.